main/src/ca/uwaterloo/flix/language/errors/ErrorCode.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2025 Magnus Madsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.language.errors
+
+class ErrorCode(val number: Int)
+
+object ErrorCode {
+
+  case object E51895 extends ErrorCode(51895) // UNUSED
```

Comment:
Don't know how we feel about OOP, but it seems easy to have a `def code(ec: ErrorCode): Int = match ec ..` function and avoid this inheritance stuff

I think calling it MismatchedArity (or whatever) is better than E51895. In this version it's hard to look at `ParserError.MismatchedArity` and see whether its correct or not. "does E51895 belong to MIsmatchedArity? who knows"

Reply:
Its simple, elegant, but a long function with no guarantee that the RHS would be unique.

Reply:
Easier to show https://github.com/flix/flix/pull/10934